Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1

```diff
@@ -147,24 +147,32 @@ if($azureWebSite) {
         deployer = 'VSTS'
         details = $buildUrl       
     })
+
+    $matchedNames = $azureWebSite.EnabledHostNames -match 'scm'
     
-    $url = [string]::Format("https://{0}.scm.azurewebsites.net/deployments/{1}",[System.Web.HttpUtility]::UrlEncode($WebSiteName),[System.Web.HttpUtility]::UrlEncode($deploymentId))
-
-    Write-Verbose "Invoke-RestMethod $url -Credential $credential  -Method PUT -Body $body -ContentType `"application/json`" -UserAgent `"myuseragent`""
-    Write-Host "Updating deployment status"
-    try {
-        Invoke-RestMethod $url -Credential $credential  -Method PUT -Body $body -ContentType "application/json" -UserAgent "myuseragent"
-    } 
-    catch {
-        Write-Verbose $_.Exception.ToString()        
-        $response = $_.Exception.Response
-        $responseStream =  $response.GetResponseStream()
-        $streamReader = New-Object System.IO.StreamReader($responseStream)
-        $streamReader.BaseStream.Position = 0
-        $streamReader.DiscardBufferedData()
-        $responseBody = $streamReader.ReadToEnd()
-        $streamReader.Close()
-        Write-Warning "Cannot update deployment status for $WebSiteName - $responseBody"        
+    if($matchedNames -and $matchedNames.count -gt 0) {
+        $WebSiteName = $matchedNames[0]
+        $url = [string]::Format("https://{0}.scm.azurewebsites.net/deployments/{1}",[System.Web.HttpUtility]::UrlEncode($WebSiteName),[System.Web.HttpUtility]::UrlEncode($deploymentId))
+
+        Write-Verbose "Invoke-RestMethod $url -Credential $credential  -Method PUT -Body $body -ContentType `"application/json`" -UserAgent `"myuseragent`""
+        Write-Host "Updating deployment status"
+        try {
+            Invoke-RestMethod $url -Credential $credential  -Method PUT -Body $body -ContentType "application/json" -UserAgent "myuseragent"
+        } 
+        catch {
+            Write-Verbose $_.Exception.ToString()        
+            $response = $_.Exception.Response
+            $responseStream =  $response.GetResponseStream()
+            $streamReader = New-Object System.IO.StreamReader($responseStream)
+            $streamReader.BaseStream.Position = 0
+            $streamReader.DiscardBufferedData()
+            $responseBody = $streamReader.ReadToEnd()
+            $streamReader.Close()
+            Write-Warning "Cannot update deployment status for $WebSiteName - $responseBody"        
+        }
+    }   
+    else {
+        Write-Warning "Cannot get azurewebsite's scm, deployment status is not updated"      
```

Comment:
Is this script shipping with Update 2? If so, all non-verbose messages should use the Get-LocalizedString command.